src/ert/storage/field_utils.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import annotations
+
+import os
+import warnings
+from collections import OrderedDict
+from dataclasses import dataclass
+from io import BytesIO
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, List, Optional, TextIO, Tuple, Union
+
+import ecl_data_io
+import numpy as np
+import roffio  # type: ignore
+from xtgeo.grid3d._gridprop_import_roff import import_roff
+
+from ert.storage.grdecl_io import export_grdecl, import_bgrdecl, read_grdecl_3d_property
```

Comment:
Some extra code is introduced here which more or less reproduces functionality that exists in `xtgeo`, I am not convinced it is worth the extra maintenance and code duplication. What is the main reason for doing so? Is it a lot faster, or uses significantly less memory?

I see the argument for `GRID`, as we are just interested in a very small subset, and it is quite memory and computationally heavy to load it through xtgeo, I am however not yet convinced that it is worth re-implementing grid properties, compared to for example suggesting improvements to the api in xtgeo.

Reply:
Where are we one this one?
I also tend to believe that it's better to extend and improve `xtgeo`, but I have not looked at the design of `xtgeo` and perhaps it's too difficult to extend it at this point.

Reply:
I think we should merge it as is just to get this done. We can make a new issue to go back to using xtgeo for field loading if we wish.

Reply:
I am fine by that Jon.